### ISSUE TITLE
Service document indexing does not work

### ISSUE THREAD

**Issue by birkland** [fcrepo4-labs/fcrepo-api-x#93]
Neither indexing nor reindexing results in any triples persisted to the triple store.  The [integration test](https://github.com/fcrepo4-labs/fcrepo-api-x/blob/master/fcrepo-api-x-integration/src/test/java/org/fcrepo/apix/integration/KarafServiceIndexingIT.java) is currently disabled due to dependency clashes that puncture the OSGi barrier between Jena 2 (from fcrepo-camel-toolbox) and Jena 3 (from api-x)

**Comment by acoburn** [comment]
I'd note that `fcrepo-camel` has been upgraded to use Jena 3 -- that change is in master and in the current snapshot. `fcrepo-camel-toolbox` has a PR that upgrades to using Jena 3 as well. After that is merged, I have one semi-significant change planned for the camel-toolbox (separate the blueprint.xml files from the implementation code to make it possible to run multiple copies of these bundles -- it also makes it possible/easier to customize the blueprint.xml configs independent of the java impl). In any case, once these changes are merged, I will cut a new release of both fcrepo-camel and fcrepo-camel-toolbox.

**Comment by acoburn** [comment]
And one last point -- fcrepo-camel is, effectively, ready for release now. I am waiting on the big PR in fcrepo-camel-toolbox to go through just in case there are any changes that need to be made to fcrepo-camel. That said, if there is a need to release fcrepo-camel/4.5.0, I can do that at a moment's notice

**Comment by birkland** [comment]
It would be nice to have API-X entirely compatible with the toolbox, especially if [it allows us](https://github.com/fcrepo4-exts/fcrepo-camel-toolbox/pull/115#issuecomment-264554219) to move API-X to Camel 2.18 as well.    I can do a smoke test with the snapshots next week (or after the semi-significant change is in) to see if any additional gremlins are found.

**Comment by ajs6f** [comment]
Please make this compatibility goodness happen! I want to be able to easily describe API-X and the toolbox as two halves of the extension Twinkie in which you can wrap your creamy sugary repo fluff for maximum tastiness, API-X facing clients and the toolbox facing other servers. But that metaphor only works with a uniform substance. Otherwise it's more like eating a hamburger with the upper bun half toasted but not the lower bun half-- a wide awake nightmare.

**Comment by acoburn** [comment]
Yes, and as soon as someone reviews and merges https://github.com/fcrepo4-exts/fcrepo-camel-toolbox/pull/115, we'll be well on our way.

**Comment by birkland** [comment]
@ajs6f @acoburn  I just created #94 to make our intention for fcrepo-camel-toolbox + API-X compatibility explicit.
